### PR TITLE
feat(ST-133): integrate meetlink add, edit and delete

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added meet link modal
- Used this modal for add and edit meet link
- Integrated with edit classroom rtk mutation in the meet link modal
- Added delete meet link function to integrate it delete confirmation modal

## Associated Ticket Link(s)

- https://trello.com/c/Z33Ud3vs

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings
Meetlink Add:

https://github.com/user-attachments/assets/eacfb353-c993-4db1-9b62-160e01703f75

Meetlink Edit:

https://github.com/user-attachments/assets/751a6fee-1eba-49fd-b8df-ce27b954caab

Meetlink Delete:

https://github.com/user-attachments/assets/c1ccbbb6-c9b9-477b-ac2b-ff364c5724dc

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
Have you added validation for the link?

**Comment by faysal-alam-sazim** [comment]
> Have you added validation for the link?

I only added validation as required field. Now I've also added validation for a valid URL.

File: modules/Classroom/components/MeetlinkModal/MeetlinkModal.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    close();
+  };
+
+  const handleAddButton = async () => {
+    if (newMeetLink.trim().length === 0) {
+      return setValidationMessage("Meet Link is required");
+    }
+
+    let formattedLink = newMeetLink.trim();
+    if (!/^https?:\/\//i.test(formattedLink)) {
```
Comment (ragib-hossain-sazim) [inline modules/Classroom/components/MeetlinkModal/MeetlinkModal.tsx:33]: [Question] can't we do this validation using Zod?

Comment (faysal-alam-sazim) [reply modules/Classroom/components/MeetlinkModal/MeetlinkModal.tsx:33]: yes we can. But I didn't use zod or react-hook-form for this one Textinput field validation.

File: modules/Classroom/components/ClassroomMeetlink/ClassroomMeetlink.tsx [hunk 2]
```diff
@@ -1,43 +1,120 @@
...
+              <ActionIcon onClick={openEditMeetlinkModal}>
+                <TbEditCircle />
+              </ActionIcon>
+              <ActionIcon onClick={openDeleteModal}>
+                <FaTrashAlt className={classes.icon} />
+              </ActionIcon>
             </>
           ) : null}
         </Flex>
+        <MeetlinkModal
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/ClassroomMeetlink/ClassroomMeetlink.tsx:84]:
```suggestion
        <AddOrEditMeetlinkModal
```

File: modules/Classroom/components/ClassroomMeetlink/ClassroomMeetlink.tsx [hunk 3]
```diff
@@ -1,43 +1,120 @@
...
+          variant="outline"
+          mb={10}
+          c={"green"}
+          color="green"
+          leftIcon={<FaPlus />}
+          onClick={openAddMeetlinkModal}
+        >
+          Add meet link
+        </Button>
+        <MeetlinkModal
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/ClassroomMeetlink/ClassroomMeetlink.tsx:112]: Same comment as above

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Pull request merged**